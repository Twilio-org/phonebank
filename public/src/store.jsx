import { applyMiddleware, createStore, combineReducers, compose } from 'redux';
import { persistStore, autoRehydrate } from 'redux-persist';
import { reducer as formReducer } from 'redux-form';

import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import immutable from 'redux-immutable-state-invariant';
import { createLogger } from 'redux-logger';

import { authStatusReducer, LOGOUT_USER } from './reducers/login';
import { accountInfoReducer } from './reducers/account_info';
import { campaignListReducer } from './reducers/campaign';
import { adminQuestionsReducer } from './reducers/admin_questions';
import { adminScriptsReducer } from './reducers/admin_scripts';
import { adminContactListsReducer } from './reducers/admin_contact_lists';

const appReducer = combineReducers({
  form: formReducer,
  auth: authStatusReducer,
  account_info: accountInfoReducer,
  admin_campaigns: campaignListReducer,
  admin_questions: adminQuestionsReducer,
  admin_scripts: adminScriptsReducer,
  admin_contact_lists: adminContactListsReducer
});

const rootReducer = (state, action) => {
  const { type } = action;
  const newState = (type === LOGOUT_USER) ? undefined : state;
  return appReducer(newState, action);
};

const middleware = [immutable(), createLogger(), promise(), thunk];

const store = createStore(rootReducer, compose(applyMiddleware(...middleware), autoRehydrate()));

persistStore(store);

// to reset the store:
// persistStore(store).purge();

export default store;
