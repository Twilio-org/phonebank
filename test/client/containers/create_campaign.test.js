import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import CreateCampaignContainer from '../../../public/src/containers/create_campaign';
import store from '../../../public/src/store';

describe('<CreateCampaignContainer />', () => {
  describe('rendering', () => {
    const routerWrappedForm = (
      <Provider store={store}>
        <MemoryRouter>
          <CreateCampaignContainer />
        </MemoryRouter>
      </Provider>
    );

    const wrapper = mount(
      routerWrappedForm
    );
    it('renders correctly', () => {
      const tree = renderer.create(
        routerWrappedForm
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
