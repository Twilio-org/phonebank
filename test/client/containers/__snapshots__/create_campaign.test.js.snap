// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateCampaignContainer /> rendering renders correctly 1`] = `
<div
  className="admin-form row"
>
  <div
    className="col-md-6 col-md-offset-3 col-xs-12"
  >
    <div
      className="page-header"
    >
      <h1>
        Add Campaign
      </h1>
    </div>
    <form>
      <div
        className="row"
      >
        <div
          className="col-xs-12"
        >
          <div
            className="row"
          >
            <div
              className="col-xs-8"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                  htmlFor={undefined}
                >
                  Name
                </label>
                <span
                  className="help-block"
                >
                  Internal
                </span>
                <input
                  className="form-control"
                  id={undefined}
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Name"
                  type=""
                  value=""
                />
              </div>
            </div>
            <div
              className="col-xs-4"
            >
              
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-xs-8"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                  htmlFor={undefined}
                >
                  Title
                </label>
                <span
                  className="help-block"
                >
                  Public
                </span>
                <input
                  className="form-control"
                  id={undefined}
                  name="title"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Title"
                  type=""
                  value=""
                />
              </div>
            </div>
            <div
              className="col-xs-4"
            >
              
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-xs-8"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                  htmlFor={undefined}
                >
                  Description
                </label>
                <textarea
                  className="form-control"
                  id={undefined}
                  name="description"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Description"
                  type=""
                  value=""
                />
              </div>
            </div>
            <div
              className="col-xs-4"
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <div
            className="form-group "
          >
            <label
              htmlFor="dropdown-basic-undefined"
            >
              Script
            </label>
            <select
              className="form-control"
              id="dropdown-basic-undefined"
              name="script_id"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              value=""
            >
              <option
                hidden={true}
              >
                Select 
                Script
              </option>
            </select>
          </div>
        </div>
        <div
          className="form-add-btn col-md-8"
        >
          <button
            className="btn btn-success"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Add Script
          </button>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <div
            className="form-group "
          >
            <label
              htmlFor="dropdown-basic-undefined"
            >
              Contact List
            </label>
            <select
              className="form-control"
              id="dropdown-basic-undefined"
              name="contact_lists_id"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              value=""
            >
              <option
                hidden={true}
              >
                Select 
                Contact List
              </option>
            </select>
          </div>
        </div>
        <div
          className="form-add-btn col-md-8"
        >
          <button
            className="btn btn-success"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Add Contact List
          </button>
        </div>
      </div>
      <hr />
      <div
        className="row"
      >
        <div
          className="col-xs-12"
        >
          <div
            className="btn-toolbar"
            role="toolbar"
          >
            <button
              className="btn btn-warning"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              Save as Draft
            </button>
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              Start Campaign
            </button>
            <button
              className="btn btn-default"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
